### ISSUE TITLE
Improving the Compare Models feature?

### ISSUE THREAD

**Issue by Mrnofish** [Akylas/oss-weather#147]
I use from time to time the above mentioned feature and there are two major annoyances I run consistently into:

The first is the choice of colors - I use the Pitch Black theme and they are generally too dark to provide optimal visibility. As an aside, picking a nicer palette would go a long way in making the app more visually pleasing, but I'd say function first and looks later.

The second is that picking some metrics appears to produce data that overlaps perfectly. This is one is twice as puzzling:

1. It may induce the impression that the app isn't plotting all the selected models, since only one of the curves is visible and there is no indication or suggestion (such as an animation, perhaps) whatsoever, that that other curves are hidden behind it.
2. It seems weird that e.g. Precipitation Chance would be exactly the same for a number of distinct models, whereas on Max or Min Temperature you'd see a reasonable spread and amount of variation; even if the delta is not huge between a model and another, the curves do not overlap so perfectly.

Another minor notation is that the metrics (hourly VS daily) are unmarked.

There's also a bunch of English labels where Proper Case is not used (not just in Compare Models, though).

**Comment by farfromrefug** [comment]
@Mrnofish thanks for the feedback. Yes it is an early feature and it has a lot of issues.
1/ yes i need to work on colors. Not easy to find the right color scheme for light/dark theme
2/ it has been reported before but i cant find a way to reproduce it. Any chance you could make a video of it?

About the last 2 issues. Any chance you can share screenshots?

**Comment by farfromrefug** [comment]
@Mrnofish  just published a new version which should hopefully fix both issues for you

**Comment by farfromrefug** [comment]
@Mrnofish is that fixed for you?

**Comment by Mrnofish** [comment]
I'm having trouble with Compare Models in 153.

While forecasts for my main location and the selected model work just fine in the main view, if I pick the same model in Compare Models, the app complains that there is no available data to display.

I could get another model to work for a while, but now that's acting up, too.

Also, it looks like it's not actually comparing anything, only data from one model gets displayed at most.

So, not sure what's going on, but it certainly doesn't seem like it's firing on all cylinders.

**Comment by farfromrefug** [comment]
@Mrnofish OK can you try 154? I made some fixes there

**Comment by Mrnofish** [comment]
Looks improved, but also like there's still issues, especially with Precipitation Probability Daily (PPD)

I couldn't get the data displayed by Compare Models to match what is shown in Main View..? I've tried several models and different locations and PPD values in Main View are completely different from those shown in the Compare Models graph.

However, testing is pretty time consuming and quite confusing since I have no idea what I'm supposed to be seeing, if there are problems with Open Meteo and/or certain models, and what the app is doing behind the scenes.

A few minor issues:

- The colors are still not very readable against a black background.
- The "MET Nordic Seamless with.." text appears truncated in the selection pane when Comparing Models.
- The "compare models" drop-down menu text is all lower-case.
- "chart" and "meteoblu" are also not using Proper Case.

**Comment by Mrnofish** [comment]
The colors are more legible now!

There are still some discrepancies in the PPD probability, though. I've found a location that has many rainy days forecast, and while most are matching, there's one that's marked as "Partly Cloudy" (no PPD is displayed), despite having a PPD of about 24% in Compare Models, whereas the previous and next day are both listed as "Light Drizzle" with their PPD being 26/18%.

There's a number of Open Meteo models that do not display anything within Compare Models, such as:

GFS-HRRR, MET Nordic, GEM Global, MeteoFrance Arpege World, MeteoFrance Arome France HD, ARPAE Seamless, ARPAE Cosmo 2I, ARPAE Cosmo 5M, NCEP NBM US Conus, JMA MSM, KNMI Harmonie Arome Netherlands.

For some models, the error message is triggered with some metrics and not others. For instance, I compiled the list above using PPD.

The ARPAE 2I and 5M models do not trigger an error with PPD, but instead do if, say, temperatureMin (Daily) is selected.

GEM Global, Arpege World, Arome France HD, ARPAE Seamless do not display any data (the model is missing from the legend on the bottom), however no error message is triggered either.

I tested this with a few disparate locations, and as far as my attempts are concerned, the problem seems location-independent.

It's hard to be very accurate, though, since the number of combinations between location, metric and model is just huge.

**Comment by farfromrefug** [comment]
@Mrnofish many models do not support all locations. Which i why you dont see any data.
If you want to confirm issues in the app data handling you can always compares with what you get here https://open-meteo.com/en/docs

**Comment by Mrnofish** [comment]
That'd be understandable, but like metnioend previously, I can get a given model/location couplet to populate the Main View, so you'd expect Compare Models to work with that same couple, right?

And I've made sure to use reasonable locations with certain models, eg. ARPAE with Italian locations, French locations for Arome and Arpege and so on.

That's also why I said the problem seems location independent.

**Comment by farfromrefug** [comment]
@Mrnofish Yes it is location dependent. What i need you test is for the location you are testing,go to https://open-meteo.com/en/docs, select all the models you are comparing in the app, and see if you get data for each of those models.

**Comment by Mrnofish** [comment]
Do me a favor, will you? Try arpae 5m with any Italian location or better yet, pick some place inside the Emilia Romagna region (for which arpae handles local forecasts), and tell me If you can see anything inside compare models?

I can get some data for that combination on the open meteo site, but the app does exactly what I wrote in the post before my last.

**Comment by farfromrefug** [comment]
@Mrnofish As i told you should first test with the Open-Meteo test page.
https://open-meteo.com/en/docs#latitude=44.4643040470969&longitude=11.120562930315515&current=&minutely_15=&hourly=temperature_2m,relative_humidity_2m&daily=weather_code,temperature_2m_max,temperature_2m_min,apparent_temperature_max,apparent_temperature_min,rain_sum,showers_sum,snowfall_sum&timezone=Europe%2FLondon&models=arpae_cosmo_5m

Locations from that region does not return anything with arpae 5m.
The app still see weather data for those locations (outside of compare) because i always ask for best_match too so that can fill the blank from specific models

https://api.open-meteo.com/v1/forecast?forecast_days=8&forecast_hours=74&forecast_minutely_15=4&hourly=precipitation_probability%2Cprecipitation%2Crain%2Cshowers%2Csnow_depth%2Csnowfall%2Cweathercode%2Cis_day%2Cuv_index%2Cwindspeed_10m%2Cwinddirection_10m%2Cwindgusts_10m%2Ccloudcover%2Capparent_temperature%2Ctemperature_2m&current=weathercode%2Cis_day%2Capparent_temperature%2Ctemperature_2m%2Cwindspeed_10m%2Cwinddirection_10m%2Cwindgusts_10m%2Ccloudcover&minutely_15=precipitation&past_days=1&daily=weathercode%2Cuv_index_max%2Cprecipitation_sum%2Cprecipitation_probability_max%2Crain_sum%2Csnowfall_sum%2Cshowers_sum%2Capparent_temperature_max%2Capparent_temperature_min%2Ctemperature_2m_max%2Ctemperature_2m_min%2Cwindspeed_10m_max%2Cwinddirection_10m_dominant%2Cwindgusts_10m_max&timeformat=unixtime&latitude=44.4643040470969&longitude=11.120562930315515&timezone=Europe%2FParis&models=best_match%2Carpae_cosmo_5m
